esp/src/ganglia/GangliaWidget.js

```diff
@@ -0,0 +1,422 @@
+                                    var graph = domConstruct.create("div", {
+                                        innerHTML: response.GraphSVGDataResponse.Graph,
+                                        class:'left'
+                                    }, tabItem.name+idx);
+                                }
+                            });
+                        });
+                    });
+                });
+            });
+        },
+
+        _onRefresh: function (id) {
+
+        },
+
+        _onFilterApply: function () {
+            var context = this;
+            var cluster = this.cluster;
+            var server = this.server;
+            var metrics = this.metrics;
+            var epochFilter = this.epochFilter;
+            var epochNow = this.epochNow;
+            var graphId = this.cluster + "_" + this.server + "_" + this.metrics + "_" + this.epochFilter;
+
+            WsRrd.GangliaRRDGraphList({
+                request:{
+                    Clusters: cluster,
+                    Servers: server,
+                    RRDMetrics: metrics,
+                    StartTime: epochFilter,
+                    EndTime: epochNow,
+                    Width: 300,
+                    Height: 120,
+                    Title: this.cluster + ":" + this.server + ":" + this.metrics
+                },
+            }).then(function (response) {
+                if (dojo.byId(graphId)) {
+                    alert(this.i18n.GraphExists);
+                } else {
+                    if (lang.exists ("GraphSVGDataResponse.Graph", response)) {
+                        var graph = domConstruct.create("div", {
+                            id: graphId,
+                            innerHTML: response.GraphSVGDataResponse.Graph,
+                            class:'left'
+                        },"graphs");
+                    }
+                }
+
+            var pMenu;
+            pMenu = new Menu({
+                targetNodeIds: [graphId]
+            });
+            pMenu.addChild(new MenuItem({
+                label: "Delete Graph",
+                onClick: function(){
+                    domConstruct.destroy(this.getParent().currentTarget);
+                }
+
+            }));
+            pMenu.addChild(new MenuItem({
+                label: "Pop Out Graph",
+                onClick: function () {
+                    context._onNewGraphPage(graphId);
+                }
+            }));
+            pMenu.startup();
+            });
+        },
+
+        _onNewGraphPage: function (graphId) {
+            xhr("esp/files/ganglia/ganglia.json", {
+                handleAs: "json"
+            }).then(function (data){
+                WsRrd.GangliaRRDGraphList({
+                request:{
+                    Clusters: cluster,
+                    Servers: server,
+                    RRDMetrics: metrics,
+                    StartTime: epochFilter,
+                    EndTime: epochNow,
```

Comment:
Aren't the following all undefined?:
+                    Clusters: cluster,
+                    Servers: server,
+                    RRDMetrics: metrics,
+                    StartTime: epochFilter,
+                    EndTime: epochNow,